src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -322,6 +340,19 @@ public override void RequestRefresh()
             }
         }
 
+        internal void TrackGetConfiguration(long methodStartTicks, string refreshReason, string operationStatus, string exceptionType)
+        {
+            IdentityModelTelemetryUtil.IncrementConfigurationManagerCounter(
+                    MetadataAddress, refreshReason, operationStatus, exceptionType);
+
+            IdentityModelTelemetryUtil.RecordTotalDuration(
+                (WatchUtility.Watch.Elapsed.Ticks - methodStartTicks) / TimeSpan.TicksPerMillisecond,
```

Comment:
Instead of performing a division each time, could we define the telemetry to send details in ticks and let the receiver perform any necessary calculations.

Reply:
The TimeProvider class has methods to directly get a TimeSpan which would clean this up